zigpy/application.py

```diff
@@ -462,6 +477,32 @@ def get_device(self, ieee=None, nwk=None):
 
         raise KeyError
 
+    def get_ephemeral_device(self, ieee, nwk) -> zigpy.device.Device:
+        """Get ephemeral device with currently unknown ieee or nwk."""
+
+        clear_expired = [d for d in self._ephemeral_devices if not d.expired]
+        self._ephemeral_devices = clear_expired
+
+        if ieee is None:
+            # search by NWK
+            for d in self._ephemeral_devices:
+                if d.nwk == nwk:
+                    return d
+
+            new_ephemeral_dev = zigpy.device.Device(self, t.EUI64([0x00] * 8), nwk)
+            self._ephemeral_devices.append(new_ephemeral_dev)
+            new_ephemeral_dev.schedule_initialize()
+            return new_ephemeral_dev
+
+        # search by IEEE
+        for d in self._ephemeral_devices:
+            if d.ieee == ieee:
+                return d
+
+        new_ephemeral_dev = zigpy.device.Device(self, ieee, t.NWK.unknown())
+        self._ephemeral_devices.append(new_ephemeral_dev)
```

Comment:
`new_ephemeral_dev.schedule_initialize()`?

Reply:
To trigger the discovery of the NWK addresses. NWK and IEEE discovery part of `Device.device_initialize()`

Reply:
Sorry, I meant to say that `schedule_initialize` isn't called for the "unknown NWK" branch but is called for the "unknown IEEE". Is intentional?